### DISCUSSION TITLE
Timing CoCalc Startup and Runtime

### DISCUSSION THREAD

**Discussion by nikkujensen** [sagemathinc/cocalc#6267] [Q&A] [answered]
Is it possible to time how long CoCalc takes to run a cell or notebook? I know you can time things in python like shown [here](https://cocalc.com/share/public_paths/8b892baf91f98d0cf6172b872c8ad6694d0f7204/notebooks/01.07-Timing-and-Profiling.ipynb). What I want to know is how long does it take to run a script when I click run. That includes the CoCalc server receiving the instructions and such.

**Comment by williamstein** [comment, marked as answer]
In the CoCalc-Jupyter notebook the time to run any cell is in the upper right corner of that cell.  See:

<img width="802" alt="image" src="https://user-images.githubusercontent.com/1276278/206359476-405d9084-94fc-474b-8df2-e1db309701d5.png">

There's no other built in functionality related to timing besides that.

**Reply by williamstein** [reply]
That's from this notebook - https://cocalc.com/wstein/support/julia-sets/files/julia-sets.ipynb
